toolchain/sem_ir/file.cpp

```diff
@@ -291,6 +291,12 @@ auto GetExprCategory(const File& file, InstId inst_id) -> ExprCategory {
       }
 
       case CARBON_KIND(BindName inst): {
+        // TODO: don't rely on value_id for expression category, since it may
```

Comment:
@zygoloid please take another look. I had to add this pretty gross hack to work around a crash bug that was exposed by a new test case in `check/testdata/class/self.carbon`.

(The changes in the other files are just routine merging as far as I know.)

Reply:
I see. I guess we would want the `BindName` inst to track whether it's in a `var` context or not, maybe with two different inst kinds, but this seems like an acceptable short-term workaround.